src/goto-programs/static_analysis.cpp

```diff
@@ -300,6 +378,28 @@ void static_analysis_baset::do_function_call_rec(
   {
     irep_idt identifier = to_symbol2t(function).get_symbol_name();
 
+    std::vector<expr2tc> new_args;
+
+    // When calling 'pthread_create' method, we need to make adjustments
+    // to properly make VSA understand the func we are going to
+    // call on the child thread
+
+    // TODO: A more robust way to find the method
+    if(has_prefix(id2string(identifier), "c:@F@pthread_create"))
+    {
+      // pthread_create(id, *attr, void *func, void *arg);
+      // From this we get the func's identifier that was called,
+      // along with the arg of that func
+      exprt tmp = migrate_expr_back(arguments[2]);
+      identifier = to_symbol_expr(tmp.op0()).get_identifier();
+
+      new_args.push_back(arguments[3]);
+
+      pthread_set.insert(identifier);
+    }
+    else
+      new_args = arguments;
```

Comment:
My previous comment still stands: we shouldn't be rewriting the function argument; why is this needed?

Reply:
I explained it in the comments above.  When we encounter `pthread_create (id, NULL, func, arg)`, the function call we need to execute is `func(arg)`, rather than `pthread_create`. Only then can we get the **value set** of `func` correctly.

That is, the argument is special compared to a normal function call, and we need to distinguish between the two cases. In subsequent functions, the argument is retrieved via `locationt`, which means that we are still getting arguments from `pthread_create`, and we need to change that.

Reply:
That's still not the way to do that in esbmc. I suggest you look into how the pthread_trampoline works in esbmc. 

Furthermore, if you need the argument to perform the analysis, this must happen during symbolic execution.